src/Microsoft.OpenApi/Validations/OpenApiValidator.cs

```diff
@@ -1,34 +1,58 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT license. 
 
+using System;
 using System.Collections.Generic;
 using System.Linq;
-using Microsoft.OpenApi.Exceptions;
 using Microsoft.OpenApi.Interfaces;
 using Microsoft.OpenApi.Models;
 using Microsoft.OpenApi.Services;
-using Microsoft.OpenApi.Validations;
 
 namespace Microsoft.OpenApi.Validations
 {
     /// <summary>
     /// Class containing dispatchers to execute validation rules on for Open API document.
     /// </summary>
-    public class OpenApiValidator : OpenApiVisitorBase
+    public class OpenApiValidator : OpenApiVisitorBase, IValidationContext 
```

Comment:
>, IValidationContext  [](start = 54, length = 21)

It's weird that the Validator is implementing the Context interface. Do we need the context interface at all?

Reply:
We could pass the entire validator object into rules, however this exposes a bunch of properties that Rules don't need access to.  The way it was implemented before was the state of the validator was refactored out into an aggregated object and the state object was passed into the Rules.  Rather than have the validator be made up of two objects, I just use the IValidationContext as a filter to limit what parts of the Validator the rules are allowed to access.

---
In reply to: [162479509](https://github.com/Microsoft/OpenAPI.NET/pull/176#discussion_r162479509) [](ancestors = 162479509)

Reply:
Ah that makes sense. Thanks for the explanation. 

I think the design itself is fine, but the nomenclature still throws me off a bit. I would expect the interface to be called something along the line of "IValidationContextHolder".

That name is mouthful though... so I'm also fine with the current state.

---
In reply to: [163074706](https://github.com/Microsoft/OpenAPI.NET/pull/176#discussion_r163074706) [](ancestors = 163074706,162479509)